packages/java/parser-jvm-plugin-nonnull/src/main/java/dev/hilla/parser/plugins/nonnull/NonnullPlugin.java

```diff
@@ -115,9 +115,43 @@ private Optional<PackageInfoModel> findClosestPackage(
                 .map(ClassInfoModel::getPackage);
     }
 
+    /**
+     * Returns a stream of all ancestor packages, starting with the immediate
+     * parent package.
+     *
+     * @param nodePath
+     *            the node path
+     * @return the stream of all ancestor packages
+     */
+    private Stream<PackageInfoModel> findAscendantPackages(
```

Comment:
I believe this method should be a part of `PackageInfoModel` to encapsulate the interaction with the Java Reflection. I guess, it's ok to return a parent package only, so this method could be called in a loop to reach the most top one.

So the method could have the following signature:
```java
public PackageInfoModel getParent();
```

Reply:
Hmm, ok, then let's have all ancestors at once. Something like that:
```java
private List<PackageInfoModel> getAncestorPackages();
```

Reply:
You mean `public`

Reply:
Oops, sure 😅

Or probably we could use `Stream` here as well 🤔

Reply:
What about implementing in `PackageInfoSourceModel`?

Reply:
Guess it's ok to have just a stub there. We need to remove them eventually because we definitely won't use them anymore